pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -344,13 +344,21 @@ def on_save_checkpoint(
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
             "dirpath": self.dirpath,
+            "best_k_models": self.best_k_models,
+            "kth_best_model_path": self.kth_best_model_path,
+            "kth_value": self.kth_value,
+            "last_model_path": self.last_model_path,
```

Comment:
I just noticed an issue with doing this.

Since we save each "ModelCheckpoint" mode sequentally, these attributes will not be correct depending on the order if more than 1 mode triggers a save for the same global step:

https://github.com/PyTorchLightning/pytorch-lightning/blob/fe940e195dceb18eb9f3bd512cea56ae3405d464/pytorch_lightning/callbacks/model_checkpoint.py#L366-L373

Currently, a "top-k" checkpoint will not include the `last_model_path` path even if it's saved right after for this global step.

I'm not sure what would be the best solution here. I think we should start recommending multiple `ModelCheckpoint` instances as a best practice because these interactions between flags can be unintuitive.

cc @awaelchli @ananthsub @jjenniferdai  
Related to #4335 and https://github.com/PyTorchLightning/pytorch-lightning/pull/11805#issuecomment-1067972348